File: backend/WebApi/Lunchorder.Domain/Dtos/UserProfile.cs
namespace Lunchorder.Domain.Dtos
{
    /// <summary>
    /// Represents the profile of the user
    /// </summary>
    public class UserProfile
    {
        /// <summary>
        /// First name of the user
        /// </summary>
        public string FirstName { get; set; }

        /// <summary>
        /// Last name of the user
        /// </summary>
        public string LastName { get; set; }

        /// <summary>
        /// Picture of the user (url)
        /// </summary>
        public string Picture { get; set; }

        /// <summary>
        /// The UI culture of the user
        /// </summary>
        public string Culture { get; set; }
    }
}

File: frontend/app/domain/dto/userProfile.ts
export class UserProfile implements app.domain.dto.IUserProfile, Serializable<UserProfile> {
    firstName: string;
    lastName: string;
    picture: string;
    culture: string;

    deserialize(input: any) : UserProfile {
        this.firstName = input.firstName;
        this.lastName = input.lastName;
        this.picture = input.picture;
        this.culture = input.culture;
        return this;
    }
}